src/haxelib/api/Installer.hx

```diff
@@ -0,0 +1,852 @@
+		try {
+			updateIfNeeded(library);
+		} catch (e:AlreadyUpToDate) {
+			userInterface.log(e.toString());
+		} catch (e:VcsUpdateCancelled) {
+			return;
+		}
+	}
+
+	/**
+		Updates all libraries in the scope.
+
+		If a library update fails, it is skipped.
+	**/
+	public function updateAll():Void {
+		final libraries = scope.getLibraryNames();
+		var updated = false;
+
+		for (library in libraries) {
+			userInterface.log('Checking $library');
+			try {
+				updateIfNeeded(library);
+				updated = true;
+			} catch(e:AlreadyUpToDate) {
+				continue;
+			} catch (e:VcsUpdateCancelled) {
+				continue;
+			} catch(e) {
+				userInterface.log("Failed to update: " + e.toString());
+				userInterface.log(e.stack.toString(), Debug);
+			}
+		}
+
+		if (updated)
+			userInterface.log("All libraries are now up-to-date");
+		else
+			userInterface.log("All libraries are already up-to-date");
+	}
+
+	function updateIfNeeded(library:ProjectName) {
+		final current = try scope.getVersion(library) catch (_:CurrentVersionException) null;
+
+		final vcsId = try VcsID.ofString(current) catch (_) null;
+		if (vcsId != null) {
+			final vcs = Vcs.get(vcsId);
+			if (vcs == null || !vcs.available)
+				throw 'Could not use $vcsId, please make sure it is installed and available in your PATH.';
+			// with version locking we'll be able to be smarter with this
+			updateVcs(library, vcsId, vcs);
+
+			scope.setVcsVersion(library, vcsId);
+
+			handleDependenciesVcs(library, vcsId, null);
+			// we dont know if a subdirectory was given anymore
+			return;
+		}
+
+		final semVer = try SemVer.ofString(current) catch (_) null;
+
+		final info = Connection.getInfo(library);
+		final library = ProjectName.ofString(info.name);
+		final latest = info.getLatest();
+
+		if (semVer != null && semVer == latest) {
+			throw new AlreadyUpToDate('Library $library is already up to date');
+		} else if (repository.isVersionInstalled(library, latest)) {
+			userInterface.log('Latest version $latest of $library is already installed');
+			// only ask if running in a global scope
+			if (!scope.isLocal && !userInterface.confirm('Set $library to $latest'))
+				return;
+		} else {
+			downloadAndInstall(library, latest);
+		}
+		scope.setVersion(library, latest);
+		userInterface.log('  Current version is now $latest');
+		userInterface.log("Done");
+
+		handleDependencies(library, latest);
+	}
+
+	function getDependencies(path:String):Dependencies {
+		final jsonPath = path + Data.JSON;
+		if (!FileSystem.exists(jsonPath))
+			return {};
+
+		return switch (Data.readData(File.getContent(jsonPath), false).dependencies) {
+			case null: {};
+			case dependencies: dependencies;
+		}
+	}
+
+	/**
+		Get the name found in the `haxelib.json` for a vcs library.
+
+		If `givenName` is an alias (it is completely different from the internal name)
+		then `givenName` is returned instead
+	 **/
+	function getName(givenName:ProjectName, id:VcsID, subDir:Null<String>):ProjectName {
```

Comment:
The name of this function doesn't suggest that this is specific to VCS libs.

Reply:
Renamed.